File: Frontend/src/types/comment.ts
export interface Comment {
  id: number;
  taskItemId: number;
  taskTitle: string;
  userId: number;
  userName: string;
  content: string;
  createdAt: string;
  updatedAt?: string | null;
}

export interface CreateCommentRequest {
  taskItemId: number;
  content: string;
}

File: Backend/CRM.API/CRM.API/Models/Comment.cs
namespace CRM.API.Models
{
    public class Comment
    {
        public int Id { get; set; }

        public int TaskItemId { get; set; }

        public int UserId { get; set; }

        public string Content { get; set; } = string.Empty;

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public DateTime? UpdatedAt { get; set; }

        public TaskItem TaskItem { get; set; } = null!;

        public User User { get; set; } = null!;
    }
}